src/server/scripts/EasternKingdoms/zone_redridge_mountains.cpp

```diff
@@ -113,147 +138,612 @@ class npc_big_earl : public CreatureScript
+                    break;
+                case SPELL_DESPAWN_KILL_CREDIT:
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, caster->GetGUID()))
+                        player->RewardPlayerAndGroupAtEvent(NPC_FOREMAN_OSLOW, player);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ETTIN_LINE_1:
+                        Talk(0);
+                        _events.ScheduleEvent(EVENT_ETTIN_LINE_2, 4s);
+                        break;
+                    case EVENT_ETTIN_LINE_2:
+                        Talk(1);
+                        _events.ScheduleEvent(EVENT_ETTIN_LINE_3, 6s);
+                        break;
+                    case EVENT_ETTIN_LINE_3:
+                        Talk(2);
+                        _events.ScheduleEvent(EVENT_MOVE_TO_WATER, 1s);
+                        break;
+                    case EVENT_MOVE_TO_WATER:
+                    {
+                        std::function<void(Movement::MoveSplineInit&)> initializer = [](Movement::MoveSplineInit& init)
+                        {
+                            Movement::PointsArray path(std::size(EttinPathToWater));
+                            std::ranges::transform(EttinPathToWater, path.begin(), [](Position pos) { return PositionToVector3(pos); });
+                            init.MovebyPath(path, 0);
+                        };
+                        me->GetMotionMaster()->LaunchMoveSpline(std::move(initializer), POINT_NEAR_WATER, MOTION_PRIORITY_NORMAL, POINT_MOTION_TYPE);
+                        break;
+                    }
+                    case EVENT_THROW_BOULDER:
+                        me->CastSpell(nullptr, SPELL_EJECT_PASSENGER_1);
+                        _events.ScheduleEvent(EVENT_PATH_AWAY, 6s);
+                        break;
+                    case EVENT_PATH_AWAY:
+                    {
+                        Talk(3);
+
+                        if (Creature* boulder = ObjectAccessor::GetCreature(*me, _boulderGUID))
+                            boulder->AI()->SetData(0, DATA_DONE);
+
+                        std::function<void(Movement::MoveSplineInit&)> initializer = [](Movement::MoveSplineInit& init)
+                        {
+                            Movement::PointsArray path(std::size(EttinPathUpHill));
+                            std::ranges::transform(EttinPathUpHill, path.begin(), [](Position const pos) { return PositionToVector3(pos); });
+                            init.MovebyPath(path, 0);
+                        };
+                        me->GetMotionMaster()->LaunchMoveSpline(std::move(initializer), POINT_UP_PATH, MOTION_PRIORITY_NORMAL, POINT_MOTION_TYPE);
```

Comment:
we could move this path into db `waypoint_path*` tables, then we just gotta do `me->GetMotionMaster()->MovePath(<pathid from db>);` and can remove any other operations done on the path in this file